support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.tsx

```diff
@@ -59,7 +55,7 @@ function getMobileSummaryTitle(
 function mapStateToProps(state: WithDeliveryCheckoutState) {
 	return {
 		fulfilmentOption: state.page.checkout.fulfilmentOption,
-		productOption: state.page.checkout.productOption,
+		productOption: state.page.checkout.productOption as ActivePaperProducts,
```

Comment:
is this typecast definitely safe?

Reply:
`ActivePaperProducts` is currently the expected prop type, what do you mean by safe? I don't think anything would change final implementation wise

Reply:
Yeah great question. So in general type casts/assertions are unsafe because we're overriding what typescript is inferring. `state.page.checkout.productOption` is of type `ProductOption` not `ActivePaperProducts`. From the code (https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/helpers/productPrice/productOptions.ts#L13) it looks like both are union types, and really `ProductOption` is a superset of `ActivePaperProducts`. So it's possible, at least from the type systems perspective, that `state.page.checkout.productOption` is one of the values that isn't in `ActivePaperProducts`. This could lead to knock on effects that typescript wouldn't be able to catch. I'd say type assertions like this are probably a code smell that indicate bigger issues with our types. Ideally we would have our types such that invalid states are actually impossible.